src/popper/runner_host.py

```diff
@@ -212,9 +212,17 @@ def _create_container(self, cid, step):
         if build:
             log.info(f"[{step.id}] docker build {img}:{tag} {build_ctx_path}")
             if not self._config.dry_run:
-                self._d.images.build(
+
+                _, build_logs = self._d.images.build(
```

Comment:
this blocks, so the output only shows until `images.build()` returns. If the output of the build is too big, this might cause memory issues. In order to avoid this, this code can use the low-level API as exemplified [here](https://stackoverflow.com/a/54715475). Very similar to what what's already here, but with a couple of tweaks.